Anastasiya Santalova/app.js

```diff
@@ -0,0 +1,363 @@
+        delButton.type = 'button';
+        delButton.id = elem.id + "-2";
+
+        if (elem.check === false) {
+          checkInput.checked = false;
+        } else {
+          checkInput.checked = true;
+        };
+
+        textLabel.innerHTML = elem.text;
+        dateSpan.innerHTML = elem.deadline;
+        delButton.innerHTML = 'X';
+
+        parentElem.appendChild(div);
+        div.appendChild(checkInput);
+        div.appendChild(textLabel);
+        div.appendChild(dateSpan);
+        div.appendChild(delSpan);
+        delSpan.appendChild(delButton);
+
+        delButtonOnclick();
+        checkboxOnclick();
+      })
+    },
+
+    filterCheck: function (check) {
+      var filteredTaskList = [];
+
+      if (check==="all") {
+        return list.show();
+      } else if (check==='do') {
+        taskList.forEach(function (curr) {
+          if (curr.check === false) {
+            filteredTaskList.push(curr);
+          }
+        });
+      } else {
+        taskList.forEach(function (curr) {
+          if (curr.check) {
+            filteredTaskList.push(curr);
+          }
+        });
+      }
+      return list.show(filteredTaskList);
+    },
+
+    filterDeadline: function (deadline) {
+      var filteredTaskList = [];
+
+      switch(deadline) {
+        case 'failed':
+          taskList.forEach(function (curr) {
+            if (compareDates(curr.deadline)[0]==="failed" || 
+                compareDates(curr.deadline)[1]==="failed") {
+              filteredTaskList.push(curr);
+            }
+          });
+          break;
+        case 'today':
+            taskList.forEach(function (curr) {
+              console.log(compareDates(curr.deadline));
+            if (compareDates(curr.deadline)[0]==="today" ||
+                compareDates(curr.deadline)[1]=== "today") {
+              filteredTaskList.push(curr);
+            }
+          });
+          break;
+        case 'tomorrow':
+            taskList.forEach(function (curr) {
+            if (compareDates(curr.deadline)[0]==="tomorrow" || 
+                compareDates(curr.deadline)[1]==="tomorrow") {
+              filteredTaskList.push(curr);
+            }
+          });
+          break;
+        case 'week':
+            taskList.forEach(function (curr) {
+            if (compareDates(curr.deadline)[0]==="week" || 
+                compareDates(curr.deadline)[1]==="week") {
+              filteredTaskList.push(curr);
+            }
+          });
+          break;
+        case 'month':
+            taskList.forEach(function (curr) {
+            if (compareDates(curr.deadline)[0]==="month" || 
+                compareDates(curr.deadline)[1]==="month") {
+              filteredTaskList.push(curr);
+            }
+          });
+          break;
+        default:
+          filteredTaskList = taskList;
+      }
+
+      return list.show(filteredTaskList);
+    },
+
+    changestat: function(id) {
+      taskList.forEach(function (curr) {
```

Comment:
I'd use `find` since it will stop after finding needed element

Reply:
As far as I know find can be used only in ES6

Reply:
Yes, you're right, my bad :)